src/KW_checkbox.c

```diff
@@ -18,35 +18,33 @@ void MouseUpCheckbox(KW_Widget * widget, int b) {
 }
 
 KW_bool KW_IsCheckboxChecked(KW_Widget * widget) {
-  KW_Checkbox * checkbox = (KW_Checkbox *) KW_GetWidgetUserData(widget);
+  KW_Checkbox * checkbox = (KW_Checkbox *) KW_GetWidgetData(widget, PaintLabel);
   return checkbox->checked;
 }
 
 KW_Widget * KW_CreateCheckbox(KW_GUI * gui, KW_Widget * parent,
                               const char * text, const KW_Rect * geometry) {
-  KW_Label *    label = calloc(sizeof(KW_Label), 1);
   KW_Checkbox * checkbox = calloc(sizeof(KW_Checkbox), 1);
   KW_Widget *   widget = KW_CreateWidget(gui, parent, geometry, PaintLabel,
```

Comment:
Even if this checkbox will only render a label, you should not use `PaintLabel` as a paint function to `KW_CreateWidget`. This paint function is going to be called when the widget tree is painted and it is going to break things - `PaintLabel` is expecting its `data` pointer to be a `KW_Label` and it is going to find a `KW_Checkbox` with either different fields or different field location.

I see below that you placed a `KW_Label` as the first struct member of `KW_Checkbox` so that it might pass as a label (effectively emulating inheritance), if inheritance was your intent, I don't see a checkbox having a `is-a` relationship with a label, its more of a `has-a` relationship. This also hinders the ability of having custom widgets as checkbox labels (I know there isn't  a lot of use cases for this, but one I can think of is using images instead of texts).

I know it is a a lot of rework on top of the fine work you're already doing but a nicer solution would be having a label just for the check icon, created internally for `KW_Checkbox`, and receiving another widget from the user to be the label itself (this would also mean that you would have to bind to geometry changes and change label geometry to match)

You should, at least, create a `PaintCheckbox` function even if it is empty and have your own userdata.

Reply:
If you think this is too much to ask, I'd be happy to work alongside.